packages/proxy/lib/http/util/prerequests.ts

```diff
@@ -230,11 +248,10 @@ export class PreRequests {
   }
 
   get (req: CypressIncomingRequest, ctxDebug, callback: GetPreRequestCb) {
-    // The initial request that loads the service worker does not get sent to CDP and it happens prior
-    // to the service worker target being added. Thus, we need to explicitly ignore it. We determine
-    // it's the service worker request via the `sec-fetch-dest` header
-    if (req.headers['sec-fetch-dest'] === 'serviceworker') {
-      ctxDebug('Ignoring request with sec-fetch-dest: serviceworker', req.proxiedUrl)
+    // The initial request that loads the service worker does not always get sent to CDP. Thus, we need to explicitly ignore it. We determine
+    // it's the service worker request via the `service-worker` header
+    if (req.headers?.['service-worker'] === 'script') {
```

Comment:
Since this is the same as above, maybe create a `hasServiceWorkerHeader` function?

Reply:
Updated: https://github.com/cypress-io/cypress/pull/28517/commits/956dd0aa92c8e7cdb0ee1284be92bdeeb52426a3